### PR TITLE
[DOC] Dir doc

### PR DESCRIPTION
**Pull request by BurdetteLamar**
The link at line 304 is fouled by the expression in line 302.  I don't know why, and I don't have a workaround.

### CODE REVIEW

File: dir.rb [hunk 1]
```diff
@@ -1,11 +1,85 @@
...
+# The closed stream may not be further manipulated,
+# and may not be reopened.
+#
+# The stream has a _position_, which is the index of an entry in the directory:
+#
+# - The initial position is zero (before the first entry).
+# - \Method #tell (aliased as #pos) returns the position.
+# - \Method #pos= sets the position (but ignores a value outside the stream),
+#   and returns the position.
+# - \Method #seek is like #pos, but returns +self+ (convenient for chaining).
```
Comment (peterzhu2118) [inline dir.rb:52]: I think `#seek` is more like `#pos=` right? It sets the position rather than gets the position.

File: dir.rb [hunk 2]
```diff
@@ -108,114 +194,157 @@ def self.open(name, encoding: nil, &block)
...
+  #    if followed by  the slash character <tt>'/'</tt>:
+  #
+  #      Dir.glob('**/').take(3) # => ["basictest/", "benchmark/", "benchmark/gc/"]
+  #
+  #    If the string pattern contains other characters
+  #    or is not followed by a slash character,
+  #    it is equivalent to <tt>'*'</tt>.
+  #
+  # - <tt>'?'</tt> Matches any single character;
+  #   similar in meaning to regexp <tt>/.{1}/</tt>:
```
Comment (peterzhu2118) [inline dir.rb:291]: Doesn't the regex `/./` mean the same as `/.{1}/`?

File: dir.rb [hunk 3]
```diff
@@ -108,114 +194,157 @@ def self.open(name, encoding: nil, &block)
...
   #
-  # Note that the pattern is not a regexp, it's closer to a shell glob.
-  # See File::fnmatch for the meaning of the +flags+ parameter.
-  # Case sensitivity depends on your system (+File::FNM_CASEFOLD+ is ignored).
+  #   Dir.glob('*') # => ["config.h", "lib", "main.rb"]
   #
-  # <code>*</code>::
-  #   Matches any file. Can be restricted by other values in the glob.
-  #   Equivalent to <code>/.*/mx</code> in regexp.
+  # with a block, calls the block with each of the _entry_names_
```
Comment (peterzhu2118) [inline dir.rb:240]: Nit: capitalization
 
```suggestion
  # With a block, calls the block with each of the _entry_names_
```

File: dir.rb [hunk 4]
```diff
@@ -108,114 +194,157 @@ def self.open(name, encoding: nil, &block)
...
-  # The results which matched single wildcard or character set are sorted in
-  # binary ascending order, unless +false+ is given as the optional +sort+
-  # keyword argument.  The order of an Array of pattern strings and braces
-  # are preserved.
+  # With no block, returns array _entry_names_;
+  # example (using the {simple file tree}[rdoc-ref:Dir@About+the+Examples]):
   #
-  # Note that the pattern is not a regexp, it's closer to a shell glob.
-  # See File::fnmatch for the meaning of the +flags+ parameter.
-  # Case sensitivity depends on your system (+File::FNM_CASEFOLD+ is ignored).
```
Comment (peterzhu2118) [inline dir.rb:153]: We should talk about the possible `flags` found in `File::Constants`.

File: dir.rb (hunk 1 above)
Comment (BurdetteLamar) [reply dir.rb]: Fixed.

File: dir.rb (hunk 3 above)
Comment (BurdetteLamar) [reply dir.rb]: Fixed.

File: dir.rb (hunk 2 above)
Comment (BurdetteLamar) [reply dir.rb]: Fixed.

File: dir.rb (hunk 4 above)
Comment (BurdetteLamar) [reply dir.rb:153]: I had not known where FNM_* were.

Comment (peterzhu2118) [reply dir.rb:153]: I think we should list the flags under `File::Constants` that can be used in this method.

Comment (peterzhu2118) [reply dir.rb:153]: Ok I guess it's fair that this isn't causing a regression

Comment (BurdetteLamar) [reply dir.rb:153]: @peterzhu2118, I will not forget about your suggestion.

Comment (BurdetteLamar) [reply dir.rb:153]: @peterzhu2118, which flags in File::Constants apply for Dir.glob?  I tried a couple of unlikelies (TRUNC, CREAT) and got no error.

Comment (peterzhu2118) [reply dir.rb:153]: It looks like invalid flags just get silently ignored.

**Pull request merged**